reporting/performance/prime_functions_performance.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Matt Borland
+//
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include "../../include/boost/math/special_functions/prime_functions.hpp"
+
+#include <benchmark/benchmark.h>
+
+template <class Z>
+void prime_sieve(benchmark::State& state)
+{
+    Z upper = static_cast<Z>(state.range(0));
+    for(auto _ : state)
+    {
+        std::vector<Z> primes;
+        benchmark::DoNotOptimize(boost::math::prime_sieve(static_cast<Z>(2), upper, std::back_inserter(primes)));
+    }
```

Comment:
You forgot the

```
state.SetComplexityN(state.range(0));
```

and so the complexity regressions are all wrong.

Reply:
@NAThompson I was hoping you would find what I had missed. I was getting a stupid result for big O, but the relationship between CPU time and sample size was as expected.